ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpRequestRetryTest.kt

```diff
@@ -361,4 +360,57 @@ class HttpRequestRetryTest {
             assertEquals(HttpStatusCode.InternalServerError, events[1].response?.status)
         }
     }
+
+    @Test
+    fun testRetryOnExceptionDoesntRetryTimeoutByDefault() = testWithEngine(MockEngine) {
+        config {
+            engine {
+                addHandler {
+                    delay(5000)
+                    fail("Should not be called")
+                }
+            }
+            install(HttpRequestRetry) {
+                retryOnException(3)
+                delay { }
+            }
+            install(HttpTimeout) {
+                requestTimeoutMillis = 1000
+            }
+        }
+
+        test { client ->
+            assertFailsWith<HttpRequestTimeoutException> {
+                client.get {}
+            }
+        }
+    }
+
+    @Test
+    fun testRetryOnExceptionRetriesTimeoutIfSet() = testWithEngine(MockEngine) {
+        config {
+            engine {
+                addHandler {
+                    delay(5000)
+                    fail("Should not be called")
+                }
+                addHandler {
+                    respondOk()
+                }
+            }
+            install(HttpRequestRetry) {
+                retryOnException(3, retryOnTimeout = true)
+                delay { }
+            }
+            install(HttpTimeout) {
+                requestTimeoutMillis = 1000
+            }
+        }
+
+        test { client ->
+            assertFailsWith<HttpRequestTimeoutException> {
```

Comment:
Shouldn't this pass? How's the expected behaviour different than the previous test?

Reply:
Seems to be correct in master :)
https://github.com/ktorio/ktor/blob/main/ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpRequestRetryTest.kt#L458